### DISCUSSION TITLE
Pre-compile for usage in other web frameworks

### DISCUSSION THREAD

**Discussion by MiguelNiblock** [GeekyAnts/NativeBase#5043] [Q&A]
Hi, I recently started using react-native with Remix framework. It's somewhat similar to Nextjs, it also does SSR, and I think it's a very promising prospect for making universal react-native apps. One obstacle I have is Remix uses the ESBuild compiler, and there is currently no way to customize the compilation from within Remix. So I though maybe if I pre-compiled native-base locally with webpack, then I could install it locally instead of relying a compilation at runtime. This would allow native-base to be used in any web framework, without special instructions for each. Currently the native-base website must offer instructions for both react and nextjs. Other frameworks might not be supported, or they would require their own compilation directions.

The nextjs directions however, are very specific to nextjs, so I'm not sure how I would perform the same steps to compile it for a generic use in mind. Could someone provide steps, or examples, or advice on how to perform this? 

Thanks, I love native-base, and want to keep using it in react, but I don't want to be limited to only nextjs, or CRA.

**Comment by sanna231** [comment]
It's a promising thing to add to our roadmap, we will definitely give this a thought. Thanks for your suggestion, will share the updates as soon as there is some development on this.
